### ISSUE TITLE
Unable to use your component with react app

### ISSUE THREAD

**Issue by encryptedcat53** [Optum/jsonschema-editor-react#2]
First of all thanks for your work, this component is exactly what i need for my project. However i've encountered an error while trying to use it.

Do you know how can i fix this problem?

Thank you

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
▼ 2 stack frames were expanded.
resolveDispatcher
node_modules/@optum/json-schema-editor/node_modules/react/cjs/react.development.js:1465
useState
node_modules/@optum/json-schema-editor/node_modules/react/cjs/react.development.js:1496
▲ 2 stack frames were expanded.
JsonSchemaEditor
src/index.tsx:39
  36 |  */
  37 | const JsonSchemaEditor = (props) => {
  38 |     const [isAdvancedOpen, setIsAdvancedOpen] = React.useState(false);
> 39 |     const [currentLens, setCurrentLens] = React.useState(new Array());
  40 |     const focusRef = React.createRef();
  41 |     const { onSchemaChange } = props;
  42 |     const onCloseImport = () => {
View compiled
▶ 18 stack frames were collapsed.
```

**Comment by tlindener** [comment]
Same problem here running React 17 with CRA

**Comment by seesharpguy** [comment]
@tlindener & @encryptedcat53 
1. Sorry for the delay in response. The control was created essentially as a proof of concept, that made it a lot further than anticipated. However, due to other priorities, it's taken me this long to coming back around to the control.
2.  I have made some adjustments to to the control
  - new state management under the hood for better performance
  - react is now a peer dependency

I have tested with CRA version 17 and worked for me locally.
